code/renderers/react/src/preset.ts

```diff
@@ -16,3 +16,26 @@ export const previewAnnotations: StorybookConfig['previewAnnotations'] = async (
     .concat([join(__dirname, 'entry-preview.mjs')])
     .concat(docsEnabled ? [join(__dirname, 'entry-preview-docs.mjs')] : []);
 };
+
+/**
+ * Try to resolve react and react-dom from the root node_modules of the project
+ * addon-docs uses this to alias react and react-dom to the project's version when possible
+ * If the user doesn't have an explicit dependency on react this will return the existing values
```

Comment:
I don't get this. The user must directly depend on `react` if `@storybook/react` is in use or do I oversee something?